clientd3d/preferences.c

```diff
@@ -0,0 +1,1684 @@
+    GetPrivateProfileString(strSection, "flip", DEF_FLIP, TCFlip, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "forward", DEF_FORWARD, TCForward, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "left", DEF_LEFT, TCLeft, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "lookdown", DEF_LOOKDOWN, TCLookdown, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "lookstraight", DEF_LOOKSTRAIGHT, TCLookstraight, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "lookup", DEF_LOOKUP, TCLookup, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "mouselooktoggle", DEF_MOUSELOOKTOGGLE, TCMouselooktoggle, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "right", DEF_RIGHT, TCRight, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "run/walk", DEF_RUNWALK, TCRunwalk, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "slideleft", DEF_SLIDELEFT, TCSlideleft, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "slideright", DEF_SLIDERIGHT, TCSlideright, nSize, strINIFile);
+
+    // Targeting
+    GetPrivateProfileString(strSection, "tabbackward", DEF_TABBACKWARD, TCTabbackward, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "tabforward", DEF_TABFORWARD, TCTabforward, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "targetclear", DEF_TARGETCLEAR, TCTargetclear, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "targetnext", DEF_TARGETNEXT, TCTargetnext, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "targetprevious", DEF_TARGETPREVIOUS, TCTargetprevious, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "targetself", DEF_TARGETSELF, TCTargetself, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "mousetarget", DEF_MOUSETARGET, TCSelecttarget, nSize, strINIFile);
+
+    // Map
+    GetPrivateProfileString(strSection, "map", DEF_MAP, TCMap, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "mapzoomin", DEF_MAPZOOMIN, TCMapzoomin, nSize, strINIFile);
+    GetPrivateProfileString(strSection, "mapzoomout", DEF_MAPZOOMOUT, TCMapzoomout, nSize, strINIFile);
+}
+
+static bool IsModifier(const char *TCValue)
+{
+    for (unsigned int i = MODIFIER_ALT; i < MODIFIER_ANY; i++)
+    {
+        if (strcmp(TCValue, TCModifiers[i]) == 0)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+static void InitModifierButton(const char *TCButtonstring, HWND modifier)
+{
+    if (IsModifier(TCButtonstring))
+    {
+        // Disable Modifier Combo
+        EnableWindow(modifier, FALSE);
+    }
+}
+
+static void AppendModifier(char *TCString, int iAppend)
+{
+    char TCAppend[128];
+    if (iAppend > MODIFIER_NONE && iAppend <= MODIFIER_ANY)
+    {
+        TCAppend[0] = '+';
+        TCAppend[1] = '\0';
+        strncat(TCAppend, TCModifiers[iAppend], sizeof(TCAppend) - strlen(TCAppend) - 1);
+    }
+    else
```

Comment:
Kind of a weird flow here.  Move the strncat from line 431 to line 425, and get rid of the else block?